tests/test_builders/test_build_linkcheck.py

```diff
@@ -100,7 +107,7 @@ def connection_count(self):
 
 
 @pytest.mark.sphinx('linkcheck', testroot='linkcheck', freshenv=True)
-def test_defaults(app):
+def test_defaults(app: Sphinx) -> None:
```

Comment:
i'm personally for this change, but it's worth noting that mypy supports pytest fixtures, and you can get IDE support for the types of test arg even without annotations, so these are not strictly required.

That said, i'm in favour of them anyway

Reply:
Thanks - I find them fairly useful when reading the source 'as-is' in plaintext editors (some of the more complicated type-hints can become distracting or take up screen space, but generally I find they're possible to narrow down to be relatively concise).

Reply:
Personally, I think having a coverage on the test function themselves is a bit too much (considering the number of test functions to add).

As such, I think it is preferrable, in a test file, to only type when you are using a non-test function (utility, fixture, etc, but not a `test_*`)é.

Now, mypy does **not** know about the type (PyCharm does not about it because I have autocompletion but `reveal_type` gives me `Any`) so it could be fine; however, your IDE could technically help you out for this.

Reply:
I'm sympathetic to all the (no pun intended) arguments.  Less typing seems good; return types shouldn't matter on `test_` methods because they shouldn't be called from other code -- on the other hand, visually and for some utilities the hints could be useful, and they could catch unintended errors (although arguably some of those will be noise).

On the principle of preferring fewer special-cases for code style, I think I'll keep the type hints on the `test_` functions.

Reply:
I think, typing the tests (since mypy doesn't know about them) could effectively be a good idea. Or, alternatively, I would suggest having a mypy plugin for that because it would really be easier (where you would complete the signatures yourself since you *know* exactly that 1) the test functions are all prefixed by `test_` and are given in a `test_*.py` file 2) the known fixtures are named the same and their type is *known*).